deploy/test/config/k8s/helm-app.yaml

```diff
@@ -0,0 +1,29 @@
+# This app is created to run end-to-end with the Secrets Provider Job
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  labels:
+    app: {{ ID }}test-env
+  name: {{ ID }}test-env
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: {{ ID }}test-env
+  template:
+    metadata:
+      labels:
+        app: {{ ID }}test-env
+    spec:
+      serviceAccountName:  {{ SERVICE_ACCOUNT }}
+      containers:
+      - image: centos:7
+        name: {{ ID }}test-app
+        command: ["sleep"]
+        args: ["infinity"]
+        env:
+        - name: TEST_SECRET
+          valueFrom:
+            secretKeyRef:
+              name: {{ K8S_SECRETS }}
```

Comment:
the value of `K8S_SECRETS` indicates the secrets that will be retrieved from conjur. i understand what you are doing here but what will happen if this variable has 2 k8s secrets? in this case it won't work.

i suggest hard-coding the secret name or creating some other variable but it's best not to use the `K8S_SECRETS` variable

Reply:
`K8S_SECRETS` is intended to be the array of k8s secrets whose value to to retrieve from conjur yes. 2 k8s secrets would be in the array format `[secret1, secret2]`. See `test-values-template.yaml`.

Reply:
yes, i know that. what i'm trying to say is that in case `K8S_SECRETS` has more than one secret the above won't work. `TEST_SECRET` is an env var in the container which will take its value from a k8s secret. you need to give the name of the secret but you are actually giving an array of secrets so it won't work.

Reply:
oh good point I understand what you are saying. I want to avoid hardcoding anything but what I can do is under the `deploy_helm_app` function is change it to `K8S_SECRET` (removing the S) and set a default value if one is not supplied
 `sed -e "s#{{ K8S_SECRET }}#${K8S_SECRET:-"test-k8s-secret"}#g" |`

Reply:
that will be ok. just don't use `K8S_SECRETS`